src/lib/Libattr/attr_fn_time.c

```diff
@@ -188,34 +188,35 @@ decode_time(struct attribute *patr, char *name, char *rescn, char *val)
  */
 /*ARGSUSED*/
 
-#define CVNBUFSZ 21
+#define CVNBUFSZ 24
 
 int
 encode_time(attribute *attr, pbs_list_head *phead, char *atname, char *rsname, int mode, svrattrl **rtnl)
 {
-	size_t	  ct;
-	char	  cvnbuf[CVNBUFSZ];
-	long 	  hr;
-	int	  min;
-	long	  n;
+	size_t ct;
+	unsigned long n;
```

Comment:
The value of attr->at_val.at_long is checked at line 209. It will not be negative. Thanks for bringing this up, because I found a bug on line 219. We should not be limiting the hour field to two digits because it represents a duration. I'll fix that.

Reply:
So what I said previously was incorrect. The %02lu directive will not truncate the hour value to two digits. However, there is no need for hours to have a field width of two if the value is less than ten. Different reason, same fix.

Reply:
Several tests rely on the hour field having two digits. That change has since been reverted.